src/process.rs

```diff
@@ -479,9 +522,9 @@ where
         unsafe { host::api::message::create_data(tag.id(), 0) };
 
         // Create reference to self
-        let this: Process<()> = Process::this();
+        let this: Process<(), S> = Process::this();
 
-        Bincode::encode(&Sendable::Shutdown(this)).unwrap();
+        S::encode(&Sendable::Shutdown(this)).unwrap();
```

Comment:
Bincode has a unique property so that it reads the exact amount of data it needs to deserialize the pointer. This doesn't work with all serializers. It's not safe to use this "serialize 2 messages one after another" trick with any serializer.

Reply:
https://github.com/lunatic-solutions/lunatic-rs/pull/64/commits/b1d102cc72e0ee6882a01e8c50e195ba427505c5